app/js/controllers/foyer/recapSituation.js

```diff
@@ -155,56 +153,20 @@ angular.module('ddsCommon').controller('RecapSituationCtrl', function($scope, $s
             $scope.hasRessources = true;
         }
 
-        $scope.ressourcesTns = [];
-        $scope.ressourcesNonTns = [];
+        $scope.ressources = [];
         ressourceTypes.forEach(function(ressourceType) {
             if ($scope.tempRessources[ressourceType.id]) {
-                $scope.ressourcesNonTns.push({
+                var ressource = {
                     type: ressourceType,
-                    totalMensuel: $scope.tempRessources[ressourceType.id].totalMensuel,
                     totalAnnuel: Math.round($scope.tempRessources[ressourceType.id].totalAnnuel)
-                });
-            }
-        });
-
-        var ressourcesMicroFiscal = {
-            type: _.find(ressourceTypes, { id: 'caMicroEntreprise' }),
-            totalAnnuel: 0
-        };
-        var ressourcesAutresTns = {
-            type: _.find(ressourceTypes, { id: 'autresRevenusTns' }),
-            totalAnnuel: 0
-        };
-        var ressourceAgricoleTns = {
-            type: _.find(ressourceTypes, { id: 'revenusAgricolesTns' }),
-            totalAnnuel: 0
-        };
-        $scope.situation.individus.forEach(function(individu) {
-            if (individu.caMicroEntreprise) {
-                ressourcesMicroFiscal.totalAnnuel += individu.caMicroEntreprise;
-                $scope.hasRessources = true;
-            }
-            if (individu.autresRevenusTns) {
-                ressourcesAutresTns.totalAnnuel += individu.autresRevenusTns;
-                $scope.hasRessources = true;
-            }
-           if (individu.revenusAgricolesTns) {
-                ressourceAgricoleTns.totalAnnuel += individu.revenusAgricolesTns;
-                $scope.hasRessources = true;
+                };
+                var ressourcesWithoutMonthlyAmount = ['caMicroEntreprise','revenusAgricolesTns','autresRevenusTns'];
```

Comment:
Good refactor.

This part should be a constant, and should be initialised outside from the loop. Here, for each resource, you are allocating a new array and three new strings; since the name is not uppercased, it also feels like it is mutable, while it isn't.

Actually, shouldn't this be something like `_.find(Resources, { isMontantAnnuel: true })`? (pseudo-code, I don't know `_`'s syntax for this)